client/lib/feed-store/index.js

```diff
@@ -89,6 +94,17 @@ function receiveFeeds( newFeeds ) {
 	} ) );
 }
 
+function discoverFeeds( data, feedUrl ) {
+	if ( typeof data !== 'undefined' ) {
+		data.feeds.map( function( feed ) {
+			if ( feed.feed_ID ) {
+				feedsByUrl[ feedUrl ] = feed.feed_ID;
+				FeedStore.emit( 'change' );
```

Comment:
Yeah, completely makes sense. Moving the `url -> id` map to a separate store really made things more organized and clean. 

Also, we no longer need to emit `change`, as the entire discovery mechanism is now changed to using Promises instead.